### ISSUE TITLE
[BUG] Segmentation Fault When Reading Fourier/Simple/Omp Angle Style

### ISSUE THREAD

**Issue by GenieTim** [lammps/lammps#2982]
**Summary**

I try to read a restart file written by Lammps. I have three examples of restart files where it does not work and fails with a Segfault when reading the angle style. It is not consistent though, as I have restart files with the same angle style which do not fail to be read. Unfortunately, they are from a different simulation (different tilt direction), so I would be thankful if I could read at least one of the three restart files.

**LAMMPS Version and Platform**

Still occurs with the current stable branch (c30ba70fab466105925d02e8ec810f5fc7f9448b, LAMMPS (29 Sep 2021)). 
Compilation on [Euler](https://scicomp.ethz.ch/wiki/Euler) with flags like this:

```bash
cmake3 -C ../cmake/presets/most.cmake -C ../cmake/presets/kokkos-openmp.cmake ../cmake \
	-DCMAKE_TUNE_FLAGS="-ftree-vectorize -ffast-math -Ofast -march=native" \
	 -DPKG_PYTHON=no -DWITH_JPEG=no \
	 -DBUILD_MPI=yes -DBUILD_OMP=yes -DLAMMPS_MACHINE=mpi \
	 -DDOWNLOAD_EIGEN3=yes \
	 -DPKG_USER-OMP=yes -DINTEL_ARCH=cpu \
	 -DPYTHON=no -DPKG_OPT=yes -DPKG_USER-TALLY=yes -DPKG_MOLECULE=yes \
	 -DPKG_USER-INTEL=yes -DPYTHON_EXECUTABLE=/usr/bin/python \
	 -DPKG_PYTHON=no -DPYTHON_EXECUTABLE=/usr/bin/python \
	 -DKokkos_ENABLE_OPENMP=yes -DBUILD_OMP=yes \
	 -DCMAKE_C_COMPILER=/cluster/apps/gcc/6.3.0/bin/gcc -DCMAKE_CXX_COMPILER=/cluster/apps/gcc/6.3.0/bin/g++ -DCMAKE_Fortran_COMPILER=/cluster/apps/gcc/6.3.0/bin/gfortran 
cmake3 --build .
```

**Expected Behavior**

I would expect using a `read_restart` or a command like `mpirun lmp_mpi -suffix omp -restart2data ./restart_M1000N159_1_cont.out ./restart_M1000N159_1_cont.dat.out` to generate a data file.

**Actual Behavior**

The output of the command listed above is:

```bash
LAMMPS (29 Sep 2021)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
using multi-threaded neighbor list subroutines
Reading restart file ...
  restart file = 8 Apr 2021, LAMMPS = 29 Sep 2021
  restoring atom style angle from restart
  triclinic box = (0.0000000 0.0000000 0.0000000) to (69.386394 69.386394 69.386394) with tilt (0.0000000 0.0000000 13.877279)
  1 by 1 by 1 MPI processor grid
  restoring pair style lj/cut/omp from restart
  restoring bond style fene/omp from restart
  restoring angle style fourier/simple/omp from restart
[eu-login-38:18144] *** Process received signal ***
[eu-login-38:18144] Signal: Segmentation fault (11)
[eu-login-38:18144] Signal code: Address not mapped (1)
[eu-login-38:18144] Failing at address: 0x2b2be036c000
[eu-login-38:18144] [ 0] /lib64/libpthread.so.0(+0xf630) [0x2b2bdb1b2630]
[eu-login-38:18144] [ 1] /cluster/home/betim/masters/lammps/build/lmp_mpi(_ZN9LAMMPS_NS7AtomVec14unpack_restartEPd+0xe36) [0xeafe86]
[eu-login-38:18144] [ 2] /cluster/home/betim/masters/lammps/build/lmp_mpi(_ZN9LAMMPS_NS11ReadRestart7commandEiPPc+0x1616) [0x874906]
[eu-login-38:18144] [ 3] /cluster/home/betim/masters/lammps/build/lmp_mpi(_ZN9LAMMPS_NS5Input15execute_commandEv+0x885) [0x7a4b25]
[eu-login-38:18144] [ 4] /cluster/home/betim/masters/lammps/build/lmp_mpi(_ZN9LAMMPS_NS5Input3oneERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x82) [0x7a59d2]
[eu-login-38:18144] [ 5] /cluster/home/betim/masters/lammps/build/lmp_mpi(_ZN9LAMMPS_NS6LAMMPSC1EiPPcP19ompi_communicator_t+0x103e) [0x7acfbe]
[eu-login-38:18144] [ 6] /cluster/home/betim/masters/lammps/build/lmp_mpi(main+0x3f) [0x777ebf]
[eu-login-38:18144] [ 7] /lib64/libc.so.6(__libc_start_main+0xf5) [0x2b2bdbdcf555]
[eu-login-38:18144] [ 8] /cluster/home/betim/masters/lammps/build/lmp_mpi() [0x77e8df]
[eu-login-38:18144] *** End of error message ***
--------------------------------------------------------------------------
mpirun noticed that process rank 0 with PID 18144 on node eu-login-38 exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```

**Steps to Reproduce**

Download the restart file listed below. Run the command listed above (`mpirun lmp_mpi -suffix omp -restart2data ./restart_M1000N159_1_cont.out ./restart_M1000N159_1_cont.dat.out`).

**Further Information, Files, and Links**

[restart_M1000N159_1_cont.out.zip](https://github.com/lammps/lammps/files/7284257/restart_M1000N159_1_cont.out.zip)

**Comment by akohlmey** [comment]
@GenieTim from that stack trace it does not look like the issue is caused by the angle style, but from reading the per-atom restart data. This looks like you could be running into a bug which may have caused that incorrect restart files were written. This bug was fixed in commit cc7538e7b92c4f7f99a778f5c5b0e9963c523c0e. If that is the case, then it will be difficult to recover the data from the restart file. However, there may be a small chance that there is another but lurking which may need fixing.

As a general rule, you should only expect restart files to be written by the same executable that wrote them. Because the file is mostly "raw" binary data, the format may change between platforms, compilers, and LAMMPS versions without LAMMPS being able to detect or correct for it.

That said, you should be able to write and read all restart files generated by the current stable version and convert them to data files. Can you provide a data file and a minimal input from a simulation where you can write but not read the restart with the 29 Sep 2021 version of LAMMPS?

**Comment by akohlmey** [comment]
P.S.: Your cmake3 command line should produce some warnings with the 29 Sep 2021 version since all USER packages were renamed or removed and their contents distributed to existing or new packages. Please see the release notes for the stable and 2Jul2021 and 28Jul2021 patch releases: https://github.com/lammps/lammps/releases

**Comment by khusnulyakin17** [comment]
Dear all

I have problem about my lammps in M2 Mac 
I install lammps in MacBook Pro M2 with home-brew. I run for calculation, I got problem like the following

[Khusnuls-MacBook-Pro:13354] *** Process received signal ***
[Khusnuls-MacBook-Pro:13354] Signal: Segmentation fault: 11 (11)
[Khusnuls-MacBook-Pro:13354] Signal code: Invalid permissions (2)
[Khusnuls-MacBook-Pro:13354] Failing at address: 0x10
[Khusnuls-MacBook-Pro:13354] [ 0] 0 libsystem_platform.dylib 0x00000001824c9a24 _sigtramp + 56
[Khusnuls-MacBook-Pro:13354] [ 1] 0 libfftw3.3.dylib 0x0000000105700724 fftw_plan_awake + 32
[Khusnuls-MacBook-Pro:13354] [ 2] 0 liblammps_mpi.0.dylib 0x00000001067e098c _ZN9LAMMPS_NS5FFT3dD2Ev + 36
[Khusnuls-MacBook-Pro:13354] [ 3] 0 liblammps_mpi.0.dylib 0x00000001067e07cc _ZN9LAMMPS_NS5FFT3dD0Ev + 12
[Khusnuls-MacBook-Pro:13354] [ 4] 0 liblammps_mpi.0.dylib 0x00000001068151ec _ZN9LAMMPS_NS4PPPM10deallocateEv + 816
[Khusnuls-MacBook-Pro:13354] [ 5] 0 liblammps_mpi.0.dylib 0x00000001068193f0 _ZN9LAMMPS_NS4PPPMD2Ev + 56
[Khusnuls-MacBook-Pro:13354] [ 6] 0 liblammps_mpi.0.dylib 0x000000010681143c _ZN9LAMMPS_NS4PPPMD0Ev + 12
[Khusnuls-MacBook-Pro:13354] [ 7] 0 liblammps_mpi.0.dylib 0x000000010640c528 _ZN9LAMMPS_NS5ForceD2Ev + 232
[Khusnuls-MacBook-Pro:13354] [ 8] 0 liblammps_mpi.0.dylib 0x00000001063f5f10 _ZN9LAMMPS_NS5ForceD0Ev + 12
[Khusnuls-MacBook-Pro:13354] [ 9] 0 liblammps_mpi.0.dylib 0x0000000106473bc4 _ZN9LAMMPS_NS6LAMMPS7destroyEv + 88
[Khusnuls-MacBook-Pro:13354] [10] 0 liblammps_mpi.0.dylib 0x00000001064738cc _ZN9LAMMPS_NS6LAMMPSD2Ev + 64
[Khusnuls-MacBook-Pro:13354] [11] 0 lmp_mpi 0x0000000104ca7db4 main + 88
[Khusnuls-MacBook-Pro:13354] [12] 0 dyld 0x00000001821190e0 start + 2360
[Khusnuls-MacBook-Pro:13354] *** End of error message ***
zsh: segmentation fault lmp_mpi -in lammps.in

Please if any solution from this problem
Thank you

**Comment by akohlmey** [comment]
@khusnulyakin17 there are several problems with your post to this issue:
- This is a closed issue. The problem has been resolved. Your post will not be directly visible on GitHub. Thus nobody will care.
- There is *no* connection of your problem to the subject of this issue.
- You don't provide any useful information to debug this issue
   - You don't specify which LAMMPS version exactly you are using?
   - You don't provide the exact configuration (e.g. output of lmp_mpi -h)
   - You don't provide your input deck
 
So please collect the information that is necessary and post a *new* bug report issue (new issues use a template with comments that have instructions what information to fill in. Also the LAMMPS manual has information for how to provide helpful additional information, specifically for segmentation faults: https://docs.lammps.org/Errors_debug.html

Bottom line, if you want help, you have to help *us* to help you *first*.
